src/darwin/Framework/CHIP/MTRDevice.mm

```diff
@@ -2144,13 +2144,9 @@ - (NSArray *)_getAttributesToReportWithReportedValues:(NSArray<NSDictionary<NSSt
             // if expected values exists, purge and update read cache
             NSArray * expectedValue = _expectedValueCache[attributePath];
             if (expectedValue) {
-                if (![self _attributeDataValue:attributeDataValue
-                            isEqualToDataValue:expectedValue[MTRDeviceExpectedValueFieldValueIndex]]) {
-                    shouldReportAttribute = YES;
-                    previousValue = expectedValue[MTRDeviceExpectedValueFieldValueIndex];
-                }
-                _expectedValueCache[attributePath] = nil;
+                previousValue = expectedValue[MTRDeviceExpectedValueFieldValueIndex];
                 _readCache[attributePath] = attributeDataValue;
+                shouldReportAttribute = NO;
```

Comment:
`shouldReportAttribute` is already NO at this point.  Why are we setting that again?

Similarly, the logging below in the !shouldReportAttribute case is completely wrong in the new setup, and will make people _extremely_ confused.

Reply:
Filed https://github.com/project-chip/connectedhomeip/issues/33013